include/athena/core/AbstractNode.h

```diff
@@ -20,34 +20,46 @@
 #include <athena/core/core_export.h>
 #include <athena/core/inner/InnerFunctions.h>
 #include <athena/core/inner/Tensor.h>
+#include <athena/core/inner/ForwardDeclarations.h>
 
 #include <string>
 #include <string_view>
+#include <map>
 
 namespace athena::core {
-using EdgeMark = size_t;
+
 
 /**
  * A Node represents a piece of computation in Graph
  */
 class ATH_CORE_EXPORT AbstractNode {
     private:
     void fullClear();
-    std::shared_ptr<inner::Tensor> mTensor;
+    std::shared_ptr<inner::Tensor> mResultTensor;
+    std::map<size_t, std::shared_ptr<inner::Tensor>> mOutgoingDerivatives;      // Index of outgoing node -> outgoing derivative for that tensor
```

Comment:
Make sure this description gets into doxygen.

```suggestion
   /// Index of outcoming node -> outcoming derivative for that tensor
    std::map<size_t, std::shared_ptr<inner::Tensor>> mOutcomingDerivatives;
```